pokemon.go

```diff
@@ -1,11 +1,110 @@
 package pokemonbattlelib
 
+import "fmt"
+
 type Pokemon struct {
-	// National Pokedex Number
-	NatDex uint16
-	// TODO
+    NatDex uint16                    // National Pokedex Number
+    Nickname string                  // player-given name for the Pokemon
+    Level Level                      // value from 1-100 influencing stats
+    Ability Ability                  // name of this Pokemon's ability
+    TotalExperience uint             // the total amount of exp this Pokemon has gained, influencing its level
+    Gender Gender                    // this Pokemon's gender
+    IVs [6]IndividualValue           // values from 0-31 that represents a Pokemon's 'genetic' potential
+    EVs [6]EffortValue               // values from 0-255 that represents a Pokemon's training in a particular stat
+    Nature Nature                    // represents a Pokemon's disposition and affects stats
+    Stats [6]Stat                    // the actual stats of a Pokemon determined from the above data
+    CurrentHP uint                   // the remaining HP of this Pokemon 
+    HeldItem Item                    // the item a Pokemon is holding
+    Moves [4]Move                    // the moves the Pokemon currenly knows
+    Friendship uint8                 // how close this Pokemon is to its Trainer
+    OriginalTrainerName string       // the name of the first Trainer who caught this Pokemon
+    OriginalTrainerID uint16         // a number associated with the first Trainer who caught this Pokemon
```

Comment:
Does this have any affect on the battle?

Reply:
Pokemon who's OTID does not match their current Trainer's gain more EXP in battle. The name might be able to be omitted though

Reply:
Let's omit the name then